recorder/src/main/cpp/controller.cc

```diff
@@ -244,53 +240,27 @@ void Controller::run_with_associate(const Buff& associate_response_buff, const T
         logger->trace("Polling now");
 
         auto next_tick = Time::now();
-        if (do_call(curl, url.c_str(), "associate-poll", retries_used, s_t_poll_rpc, s_c_poll_rpc_failures)) {
-            accept_work(recv, host, port);
-            backoff_seconds = cfg.backoff_start;
-            retries_used = 0;
+        if (do_call(curl, poll_url.c_str(), "associate-poll", poll_retries_used, s_t_poll_rpc, s_c_poll_rpc_failures)) {
+            accept_work(recv, poll_host, poll_port);
+            poll_backoff_seconds = cfg.backoff_start;
+            poll_retries_used = 0;
             next_tick += Time::sec(cfg.poll_itvl);
             poll_tick = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now().time_since_epoch()).count();
         } else {
-            if (retries_used++ >= cfg.max_retries) {
-                logger->error("COMM failed too many times, giving up on the associate: {}", url);
+            if (poll_retries_used++ >= cfg.max_retries) {
+                // old associate seems gone for good, let us find a new one
+                logger->error("COMM failed too many times, giving up on the associate: {}", poll_url);
+                scheduler.schedule(Time::now(), assoc_cb);
```

Comment:
In case of too many polls, if the assoc_cb succeeds then the next poll will have previous values in "poll_backoff_seconds", "poll_retries_used".